File: 1.21.6/src/main/java/com/bleudev/nine_lifes/util/TextUtils.java
package com.bleudev.nine_lifes.util;

import net.minecraft.text.ClickEvent;
import net.minecraft.text.MutableText;
import net.minecraft.text.Text;

import java.net.URI;

public class TextUtils {
    public static MutableText link(MutableText text, String uri) {
        return text.setStyle(text.getStyle().withClickEvent(new ClickEvent.OpenUrl(URI.create(uri))));
    }
    public static MutableText link(String uri) {
        return literal_link(uri, uri);
    }

    public static MutableText literal_link(String string, String uri) {
        return link(Text.literal(string), uri);
    }
}

File: 26.3/src/main/kotlin/com/bleudev/nine_lifes/util/TextUtils.kt
package com.bleudev.nine_lifes.util

import net.minecraft.network.chat.ClickEvent
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.MutableComponent
import java.net.URI

fun link(text: MutableComponent, uri: String): MutableComponent = text
    .setStyle(text.style.withClickEvent(ClickEvent.OpenUrl(URI.create(uri))))
fun link(uri: String): MutableComponent = literal_link(uri, uri)
fun literal_link(string: String, uri: String): MutableComponent = link(Component.literal(string), uri)